src/axom/inlet/Table.cpp

```diff
@@ -92,73 +151,65 @@ std::shared_ptr<Field> Table::addField(axom::sidre::Group* sidreGroup,
 }
 
 std::shared_ptr<Field> Table::addBool(const std::string& name,
-                                      const std::string& description)
-{
+                                      const std::string& description, 
+                                      bool forArray, bool num) {
   std::string fullName = appendPrefix(m_name, name);
   axom::sidre::Group* sidreGroup = createSidreGroup(fullName, description);
-  if (sidreGroup == nullptr)
-  {
+  if (sidreGroup == nullptr) {
     //TODO: better idea?
     return std::shared_ptr<Field>(nullptr);
   }
 
-  bool value;
-  if(m_reader->getBool(fullName, value))
-  {
+  bool value = num;
+  if (forArray || m_reader->getBool(fullName, value)) {
     sidreGroup->createViewScalar("value", value? int8(1) : int8(0) );
   }          
   return addField(sidreGroup, axom::sidre::DataTypeId::INT8_ID, fullName, name);         
 }
 
 std::shared_ptr<Field> Table::addDouble(const std::string& name,
-                                        const std::string& description)
-{
+                                        const std::string& description,
+                                        bool forArray, double num) {
```

Comment:
I'm confused by the new parameters.  Can you explain why they are needed? Are you expecting users to ignore them? If not, maybe move this to an internal function the users wont see.

Reply:
These are needed for when the addArray functions add the individual elements as fields. I would expect users to ignore them.

Reply:
Then we should move to an internal function that the users dont see.